index.tsx

```diff
@@ -115,7 +113,7 @@ export const ThemeProvider: React.FC<ThemeProviderProps> = ({
 
   useEffect(() => {
     const handleStorage = (e: StorageEvent) => {
-      if (e.key !== 'theme') {
+      if (e.key !== storageKey) {
```

Comment:
Fixes another bug in which localStorage syncing wouldn't work if specifying a custom `storageKey`